src/osemgrep/language_server/Test_LS_e2e.ml

```diff
@@ -1024,24 +1025,28 @@ let test_ls_libev () =
 (* Entry point *)
 (*****************************************************************************)
 
+module Test = Alcotest_ext
+
 let promise_tests =
-  ignore test_login;
   [
-    ("Test LS", test_ls_specs);
-    ("Test LS exts", test_ls_ext);
-    ("Test LS multi-workspaces", test_ls_multi);
-    (* TODO: currently failing in js tests in CI
-          ("Test Login", test_login);
-    *)
-    ("Test LS with no folders", test_ls_no_folders);
+    Test.create "Test LS" test_ls_specs ~tolerate_chdir:true;
+    Test.create "Test LS exts" test_ls_ext ~tolerate_chdir:true;
+    Test.create "Test LS multi-workspaces" test_ls_multi ~tolerate_chdir:true;
+    Test.create "Test LS login" test_login
+      ~expected_outcome:
+        (Should_fail "TODO: currently failing in js tests in CI");
+    Test.create "Test LS with no folders" test_ls_no_folders;
   ]
-  |> List_.map (fun (s, f) -> (s, with_timeout f))
+  |> List_.map (fun (test : _ Test.t) ->
+         Test.update test ~func:(with_timeout test.func))
 
 let tests =
   let prepare f () = Lwt_platform.run (f ()) in
-  Alcotest_ext.pack_tests "Language Server (e2e)"
-    (promise_tests |> List_.map (fun (s, f) -> (s, prepare f)))
+  Test.pack_tests_pro "Language Server (e2e)"
```

Comment:
what is this pack_tests vs pack_tests_pro? pro does not ring a bell to me (or I think about semgrep-pro, which
is probably not intended).

Reply:
Ha. This was to minimize the changes I had to make originally in the semgrep code base. `pack_tests` works on the pairs (name, func). This hasn't changed. `pack_tests_pro` is the same but works on the new test type. `Alcotest.pack_tests_pro` should replace `Alcotest.pack_tests`.

Reply:
I'll change this soon-ish.